new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSessionMigration.java

```diff
@@ -0,0 +1,446 @@
+    final String webServiceGetUrl = SESSMIGR_APP_WAR_NAME + "/?getCounter";
+
+    logger.info("Got the primary server: {0}, the secondary server: {1} "
+        + "and session create time: {2} from the factory method getOrigServerAndSessionInfo",
+            origPrimaryServerName, origSecondaryServerName, origSessionCreateTime);
+
+    // stop the primary server by changing ServerStartPolicy to NEVER and patching domain
+    logger.info("Shut down the primary server: {0}", origPrimaryServerName);
+    shutdownServerUsingServerStartPolicy(origPrimaryServerName);
+
+    // get HTTP response data from web service deployed on the cluster
+    Map<String, String> httpDataInfo =
+        processHttpRequest(domainUid + "-" + origSecondaryServerName,
+            webServiceGetUrl, " -b ");
+    String primaryServerName = httpDataInfo.get(primaryServerAttr);
+    String sessionCreateTime = httpDataInfo.get(sessionCreateTimeAttr);
+    String countStr = httpDataInfo.get(countAttr);
+
+    // verify that the HTTP response data are not null
+    assertAll("Check that WebLogic server and session vars is not null or empty",
+        () -> assertNotNull(primaryServerName,"Primary server name shouldn’t be null"),
+        () -> assertNotNull(sessionCreateTime,"Session create time shouldn’t be null"),
+        () -> assertNotNull(countStr,"Session state shouldn’t be null")
+    );
+
+    int count = Optional.ofNullable(countStr).map(Ints::tryParse).orElse(0);
+
+    logger.info("After patching the domain, the primary server changes to : {0} "
+        + ", session create time: {1} and session state: {2}",
+            primaryServerName, sessionCreateTime, countStr);
+
+    // verify that a new primary server is picked and HTTP session state is migrated
+    assertAll("Check that WebLogic server and session vars is not null or empty",
+        () -> assertNotEquals(origPrimaryServerName, primaryServerName,
+            "After the primary server stopped, another server should become the new primary server"),
+        () -> assertEquals(origSessionCreateTime, sessionCreateTime,
+            "After the primary server stopped, HTTP session state should be migrated to the new primary server"),
+        () -> assertEquals(count, SESSION_STATE,
+            "After the primary server stopped, HTTP session state should be migrated to the new primary server")
+    );
+
+    logger.info("SUCCESS --- testSessionMigration \nThe new primary server is : {0}, it was : {1}. "
+        + "\nThe session state was set to {2}, it is migrated to the new primary server.",
+            primaryServerName, origPrimaryServerName, SESSION_STATE);
+  }
+
+  /**
+   * A factory method generates a stream of String and referred
+   * by the parameterized test method testSessionMigration.
+   *
+   * @return a stream of Arguments that contains primary and secondary server names and session create time
+   */
+  static Stream<Arguments> getOrigServerAndSessionInfo() {
```

Comment:
change the method name to getOriginalServerAndSessionInfo

Reply:
changed